### PR TITLE
Accessibility

### PR DESCRIPTION
**Pull request by SanderGi**
## What did I do?
- keyboard navigation (feature)
- focus outline (styles)
- aria roles (aria-selected, listbox, option) on slices
- fixed all vulnerabilities
- refactored into using FormattingService

## Why did I do it this way?
The appropriate aria roles and labels make the visual more understandable to screen readers which makes the sunburst chart viewable to blind users and more. Keyboard navigation allows these users to actually use the sunburst chart.

## Anything else you should know?
This PR comes with ~20 extra test cases for the new aria attributes and keyboard navigation.

## Screenshots
Aria attributes being set:
image

Keyboard navigation:
Recording 2023-07-14 at 14 08 01

Formatting model using settings service:
image

### CODE REVIEW

File: src/behavior.ts [hunk 1]
```diff
@@ -114,5 +110,18 @@ export class Behavior implements IInteractiveBehavior {
                 !selected && hasHighlights
             );
         });
+
+        selection.attr("aria-selected", (dataPoint: HierarchyRectangularNode<SunburstDataPoint>) => {
+            return dataPoint.data.selected;
+        });
+
+        selection.on("keydown", (e, d: HierarchyRectangularNode<SunburstDataPoint>) => {
```
Comment (FirzinatKhuzeev) [inline src/behavior.ts:118]: Extract to `bindEvents` sections

File: src/behavior.ts [hunk 2]
```diff
@@ -114,5 +110,18 @@ export class Behavior implements IInteractiveBehavior {
...
+            // if keydown is not enter or space, return
+            if (e.keyCode !== 13 && e.keyCode !== 32) {
```
Comment (FirzinatKhuzeev) [inline src/behavior.ts:120]: Use [code ](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code) instead of keyCode

nit: you can add constants for codes.

File: src/Sunburst.ts [hunk 3]
```diff
@@ -220,7 +212,10 @@ export class Sunburst implements IVisual {
         this.selectionManager = options.host.createSelectionManager();
 
         this.main = this.svg.append("g");
-        this.main.attr(CssConstants.transformProperty, translate(Sunburst.CentralPoint, Sunburst.CentralPoint));
+        this.main
+            .attr(CssConstants.transformProperty, translate(Sunburst.CentralPoint, Sunburst.CentralPoint))
+            .attr("role","listbox")
+            .attr("aria-multiselectable","true");
```
Comment (FirzinatKhuzeev) [inline src/Sunburst.ts:218]: Auto format document (Shift + Alt + F)

File: src/Sunburst.ts [hunk 4]
```diff
@@ -442,6 +712,10 @@ export class Sunburst implements IVisual {
         this.renderTooltip(pathSelectionMerged);
         this.setCategoryLabelPosition(this.viewport.width);
         this.setPercentageLabelPosition(this.viewport.width);
+        this.main.selectAll("path." + this.appCssConstants.slice.className)
+            .attr("role","option")
+            .attr("tabindex","0");
+        this.main.selectAll("text").attr("role","presentation");
```
Comment (FirzinatKhuzeev) [inline src/Sunburst.ts:718]: move `.attr("role", "presentation")` near to `.append("textPath")` because it is expensive operation.

Comment (FirzinatKhuzeev) [reply src/Sunburst.ts:718]: add aria-label lke this: `.attr('aria-label', (d: HierarchyRectangularNode<SunburstDataPoint>) => d.data.name);` 

Screen reader will announce value for list items.

File: src/behavior.ts (hunk 2 above)
Comment (SanderGi) [reply src/behavior.ts]: Done: [6f3e3c3](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/6f3e3c3a495dab6c497e51b353a189744d85cdc9)

File: src/behavior.ts [hunk 5]
```diff
@@ -114,5 +110,18 @@ export class Behavior implements IInteractiveBehavior {
...
+                return;
+            }
+            console.log(document.activeElement);
```
[the remark this answers is not in the document]
Comment (SanderGi) [reply src/behavior.ts]: removed consol.log: [6f3e3c3](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/6f3e3c3a495dab6c497e51b353a189744d85cdc9)

[the remark this answers is not in the document]
Comment (SanderGi) [reply src/behavior.ts]: onSelect and handleSelection: [13f944f](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/13f944f4a01d13ab0d0643a6646d191a61a7e2ca)

File: src/behavior.ts (hunk 1 above)
Comment (SanderGi) [reply src/behavior.ts]: done: [13f944f](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/13f944f4a01d13ab0d0643a6646d191a61a7e2ca)

File: src/Sunburst.ts (hunk 3 above)
Comment (SanderGi) [reply src/Sunburst.ts]: done: [6f2c55a](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/6f2c55a2b93500fb52cd3474ca6ff81199a675db)

File: src/Sunburst.ts (hunk 4 above)
Comment (SanderGi) [reply src/Sunburst.ts]: Done: [57cc2e5](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/57cc2e5bf79b2464761d36e369c7fb0138cc4c54)

**Comment by SanderGi** [comment]
@zBritva when we changed "Show Category Label" to "Show Selected Category Label" we never added translations for this in /stringResources (also I think I might have disabled the translation for some of the components). Is it acceptable to use google translate to fill in the new translations for the new controls?

**Comment by zBritva** [comment]
image

Need to add `aria-label` and `role` attributes for unselected state of the visual (When user selects category the attributes works)

**Comment by SanderGi** [comment]
> Need to add `aria-label` and `role` attributes for unselected state of the visual (When user selects category the attributes works)

Weird, they're there in the html initially: 
image
image

Can you link the speech viewer you're using? I'll try to figure out why there is a discrepancy

**Comment by SanderGi** [comment]
I tested the attributes with Mac's build in VoiceOver using the powerbi-dektop sample data and all the attributes work

**Comment by FirzinatKhuzeev** [comment]
> > Need to add `aria-label` and `role` attributes for unselected state of the visual (When user selects category the attributes works)
> 
> Weird, they're there in the html initially: image image
> 
> Can you link the speech viewer you're using? I'll try to figure out why there is a discrepancy
> 
> Also, if you have time, please let me know what the tests give you. They should already check if these attributes work in the unselected state: image

can you point me to `aria-label` in src please?

**Comment by SanderGi** [comment]
> > > Need to add `aria-label` and `role` attributes for unselected state of the visual (When user selects category the attributes works)
> > 
> > 
> > Weird, they're there in the html initially: image image
> > Can you link the speech viewer you're using? I'll try to figure out why there is a discrepancy
> > Also, if you have time, please let me know what the tests give you. They should already check if these attributes work in the unselected state: image
> 
> can you point me to `aria-label` in src please?

https://github.com/microsoft/powerbi-visuals-sunburst/blob/d9c073ca8b6c02754ea2589534ede68f2184cc4d/src/Sunburst.ts#L719C9-L719C9

File: src/Sunburst.ts [hunk 6]
```diff
@@ -327,53 +322,325 @@ export class Sunburst implements IVisual {
...
+        const group_group: powerbi.visuals.FormattingGroup = this.getGroupGroup();
+        visual_datacard.groups.push(group_group);
+
+        const tooltip_group: powerbi.visuals.FormattingGroup = this.getTooltipGroup();
+        visual_datacard.groups.push(tooltip_group);
+
+        const legend_group: powerbi.visuals.FormattingGroup = this.getLegendGroup();
+        visual_datacard.groups.push(legend_group);
+
+        return { cards: [visual_datacard] };
```
Comment (MulyukovAidar) [inline src/Sunburst.ts:342]: consider using 
https://github.com/microsoft/powerbi-visuals-utils-formattingmodel/blob/main/src/FormattingSettingsService.ts#L49C26-L49C26

File: test/visualTest.ts [hunk 7]
```diff
@@ -229,6 +232,171 @@ describe("Sunburst", () => {
...
+            dataView.metadata.objects = {
+                group: { labelFontSize: fontSize }
+            };
+            visualBuilder.updateRenderTimeout(
+                dataView,
+                () => {
+                    const firstClickPoint: HTMLElement = visualBuilder.slices[visualBuilder.slices.length - 1];
+                    const secondClickPoint: HTMLElement = visualBuilder.mainElement[0];
+                    d3Click(firstClickPoint, 5, 5);
+                    setTimeout(
```
Comment (MulyukovAidar) [inline test/visualTest.ts:253]: something like this might be handy here.    
   function timeout(ms: number) {
        return new Promise(resolve => setTimeout(resolve, ms));
    }

File: test/visualTest.ts [hunk 8]
```diff
@@ -229,6 +232,171 @@ describe("Sunburst", () => {
...
+                                DefaultWaitForRender);
+                        },
+                        DefaultWaitForRender);
+                },
+                2,
+                DefaultWaitForRender);
+        });
+
+        it("data label font weight should be correct", (done: DoneFn) => {
+            dataView = defaultDataViewBuilder.getDataView(
```
Comment (MulyukovAidar) [inline test/visualTest.ts:270]: use beforeEach() for repetitive code in tests

File: test/visualTest.ts [hunk 9]
```diff
@@ -372,13 +740,16 @@ describe("Sunburst", () => {
...
-                        instance.properties["fill"] &&
-                        instance.properties["fill"]["solid"] &&
-                        instance.properties["fill"]["solid"]["color"] &&
-                        instance.properties["fill"]["solid"]["color"] === color);
+                    const result: powerbi.visuals.FormattingModel = visualBuilder.instance.getFormattingModel();
+                    const visual_card: powerbi.visuals.FormattingCard = <powerbi.visuals.FormattingCard>result.cards[0];
+                    const group_group: powerbi.visuals.FormattingGroup = <powerbi.visuals.FormattingGroup>visual_card.groups[0];
+                    const group_slices: powerbi.visuals.FormattingSlice[] = <powerbi.visuals.FormattingSlice[]>group_group.slices;
+                    const colorExist: boolean = group_slices.some((slice: powerbi.visuals.FormattingSlice) =>
+                        slice["control"] &&
```
Comment (MulyukovAidar) [inline test/visualTest.ts:748]: why not 
slice?.control?.properties?.value?.value === color

File: src/behavior.ts [hunk 10]
```diff
@@ -67,6 +65,23 @@ export interface BehaviorOptions extends IBehaviorOptions<SunburstDataPoint> {
...
+        if (onSelect) {
+            onSelect(d.data);
+        }
+    }
+
+    private clear(selectionHandler: ISelectionHandler, onSelect: (dataPoint: SunburstDataPoint) => void) {
+        selectionHandler.handleClearSelection();
+
+        if (onSelect) {
+            onSelect(null);
```
Comment (MulyukovAidar) [inline src/behavior.ts:81]: please explain this onSelect call

**Review by MulyukovAidar — CHANGES_REQUESTED** [review]

File: package.json [hunk 11]
```diff
@@ -71,13 +73,18 @@
...
     "ts-loader": "8.0.12",
     "ts-node": "^9.1.1",
     "tslint": "5.20.1",
     "tslint-microsoft-contrib": "6.2.0",
     "typescript": "4.1.3",
     "webpack": "^4.44.2"
   },
-  "dependencies": {}
+  "dependencies": {
+    "@typescript-eslint/eslint-plugin": "^5.60.1",
```
[the remark this answers is not in the document]
Comment (SanderGi) [reply package.json:83]: Done: [c58e389](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/c58e38971ff084ff01d3dcd16cba375f9d7c8269)

File: src/SunburstSettings.ts [hunk 12]
```diff
@@ -28,9 +28,19 @@ import { dataViewObjectsParser } from "powerbi-visuals-utils-dataviewutils";
 import DataViewObjectsParser = dataViewObjectsParser.DataViewObjectsParser;
 
 class SunburstGroupSettings {
-    public fontSize: number = 14;
     public showSelected: boolean = true;
+    public fontSize: number = 14;
```
[the remark this answers is not in the document]
Comment (SanderGi) [reply src/SunburstSettings.ts:32]: There is only one source of truth, this class is where the formatting settings come from:
image
image

File: test/visualTest.ts (hunk 7 above)
Comment (SanderGi) [reply test/visualTest.ts]: I use that function for the new test cases but did not want to modify the old test cases. I have modified the old test cases to use that function: [615bcac](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/615bcac76a6f2f971f46984c5d39e28ebce56a7f)

File: test/visualTest.ts (hunk 8 above)
Comment (SanderGi) [reply test/visualTest.ts]: Done: [7a120ff](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/7a120ff3160317fefdf7313f99f253101efb7f43)

File: test/visualTest.ts (hunk 9 above)
Comment (SanderGi) [reply test/visualTest.ts]: Sure: [14f3b24](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/14f3b244ca97294037b1aa42304101f4b07cef28)

File: src/behavior.ts (hunk 10 above)
Comment (SanderGi) [reply src/behavior.ts:81]: It does the same thing as it did in the previous code before being refactored into the clear() function:

It let's the visual know to render without anything in the selected state.

File: src/Sunburst.ts (hunk 6 above)
Comment (SanderGi) [reply src/Sunburst.ts]: This would likely make the automated slice colorpicker code more complicated: 
image

And would be quite a lengthy refactor, which I believe might be beyond the scope of an accessibility focused PR?

What are the advantages of using the FormattingSettingsService library other than managing uids automatically which can be done with any uid function? I can move the formatting model code to a separate file if the idea is to keep the model specification separate from the visual logic?

File: capabilities.json [hunk 13]
```diff
@@ -82,45 +124,29 @@
...
             "numeric": true
           }
         }
       }
     },
     "legend": {
-      "displayName": "Legend",
-      "description": "Display legend options",
-      "descriptionKey": "Visual_Description_Legend",
-      "displayNameKey": "Visual_DisplayLegendOptions",
```
Comment (uve) [inline capabilities.json:110]: Do not remove displayNameKey strings

File: package.json [hunk 14]
```diff
@@ -71,13 +73,16 @@
...
+    "stylelint": "^14.0.0",
     "ts-loader": "8.0.12",
     "ts-node": "^9.1.1",
     "tslint": "5.20.1",
     "tslint-microsoft-contrib": "6.2.0",
     "typescript": "4.1.3",
-    "webpack": "^4.44.2"
-  },
-  "dependencies": {}
+    "webpack": "^4.44.2",
```
Comment (uve) [inline package.json:82]: remove webpack

File: src/Sunburst.ts [hunk 15]
```diff
@@ -327,53 +322,325 @@ export class Sunburst implements IVisual {
...
+        visualDatacard.groups.push(legendGroup);
+
+        return { cards: [visualDatacard] };
+    }
+
+    // eslint-disable-next-line max-lines-per-function
+    private getGroupGroup(): powerbiVisualsApi.visuals.FormattingGroup {
+        const groupGroup: powerbi.visuals.FormattingGroup = {
+            displayName: "Group",
+            uid: "visual_datacard_group_group_uid",
```
Comment (uve) [inline src/Sunburst.ts:349]: why do you define uid here?

File: src/Sunburst.ts [hunk 16]
```diff
@@ -559,14 +833,6 @@ export class Sunburst implements IVisual {
         data.total += newDataPointNode.value;
         newDataPointNode.children = [];
 
-        if (level === 1 && originParentNode.children.length > 0) {
```
Comment (uve) [inline src/Sunburst.ts:562]: Why are you removing these lines?

**Review by uve — CHANGES_REQUESTED** [review]

File: capabilities.json (hunk 13 above)
Comment (SanderGi) [reply capabilities.json:110]: I believe I was told to remove these here: https://github.com/microsoft/powerbi-visuals-heatmap/pull/118#discussion_r1276455872. Which one is correct?

File: src/Sunburst.ts (hunk 15 above)
Comment (SanderGi) [reply src/Sunburst.ts:349]: This way of defining the formatting model requires a uid: https://learn.microsoft.com/en-us/power-bi/developer/visuals/format-pane

File: package.json (hunk 14 above)
Comment (SanderGi) [reply package.json]: Done: [4e8c27c](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/4e8c27c53faa1a6bf615d731086633bf0d1652c4)

File: src/Sunburst.ts (hunk 16 above)
Comment (SanderGi) [reply src/Sunburst.ts:562]: Because they don't do anything (their variables are never used) so the linter will complain if they are not removed

File: capabilities.json (hunk 13 above)
Comment (MulyukovAidar) [reply capabilities.json:110]: @uve there is no need to keep displayNameKey here

File: src/SunburstSettings.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (MulyukovAidar) [reply src/SunburstSettings.ts:32]: There is a better way to populate the model, Ilfat has the details

File: src/Sunburst.ts (hunk 15 above)
Comment (MulyukovAidar) [reply src/Sunburst.ts:349]: use formattingSettingsService for this

File: src/Sunburst.ts (hunk 6 above)
Comment (MulyukovAidar) [reply src/Sunburst.ts]: >What are the advantages of using the FormattingSettingsService library other than managing uids automatically which can be done with any uid function?
1) Model structure might change over time, there is no need to rely on details as the service does it for you
2) The service handles locales automatically (if has corresponding displayNameKeys)
> I can move the formatting model code to a separate file if the idea is to keep the model specification separate from the visual logic?
Yes, separating formatting model into a file is a good idea.

File: capabilities.json (hunk 13 above)
Comment (SanderGi) [reply capabilities.json:110]: Resolving since this does not appear to be an issue and will for sure go away once the formatting service is used instead

File: src/Sunburst.ts (hunk 15 above)
Comment (SanderGi) [reply src/Sunburst.ts:349]: Ok, I'll add the formatting service as per the other review

File: src/Sunburst.ts (hunk 6 above)
Comment (SanderGi) [reply src/Sunburst.ts]: [ae5b160](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/ae5b1607cd996a3127a70f8ebd29835e145cf310)

**Comment by SanderGi** [comment]
Fixed all vulnerabilities 
image

**Comment by SanderGi** [comment]
The code is made to run with the latest long term stable release of node (currently v18.17.0). Should I update the github workflow to work with this version?
image

**Comment by uve** [comment]
@SanderGi please make a rebase

**Comment by SanderGi** [comment]
Merged in changes from https://github.com/microsoft/powerbi-visuals-sunburst/pull/94 and closed that PR [following this comment](https://github.com/microsoft/powerbi-visuals-sunburst/pull/94#issuecomment-1666214330)

**Comment by SanderGi** [comment]
> @SanderGi please make a rebase

This branch should now be up to date with the dev branch

File: karma.conf.ts [hunk 17]
```diff
@@ -34,7 +34,9 @@ const testRecursivePath = "test/visualTest.ts";
 const srcOriginalRecursivePath = "src/**/*.ts";
 const coverageFolder = "coverage";
 
-process.env.CHROME_BIN = require("playwright").chromium.executablePath();
+process.env.CHROME_BIN = require("puppeteer").executablePath();
+
+//import { Config, ConfigOptions } from "karma";
```
Comment (AleksSavelev) [inline karma.conf.ts:39]: Why is it commented?

File: package.json [hunk 18]
```diff
@@ -23,54 +23,62 @@
...
-    "ts-node": "^10.9.1",
-    "typescript": "5.1.6",
-    "webpack": "^5.88.2"
+    "puppeteer": "^5.5.0",
+    "regenerator-runtime": "^0.13.7",
+    "style-loader": "^2.0.0",
+    "stylelint": "^14.16.1",
+    "ts-loader": "^8.0.12",
+    "ts-node": "^10.8.1",
+    "typescript": "^4.1.3"
```
Comment (AleksSavelev) [inline package.json:82]: Please, update to 4.9.5

File: package.json [hunk 19]
```diff
@@ -23,54 +23,62 @@
...
+    "karma-webpack": "^5.0.0",
+    "less": "^4.0.0",
+    "less-loader": "^7.2.1",
+    "lodash-es": "^4.17.20",
     "postcss-less": "^6.0.0",
-    "powerbi-models": "^1.12.6",
+    "powerbi-models": "^1.7.0",
     "powerbi-visuals-api": "~5.4.0",
-    "powerbi-visuals-tools": "5.0.2",
+    "powerbi-visuals-tools": "^5.0.1",
```
Comment (AleksSavelev) [inline package.json:65]: Please, update to 5.0.2

File: package.json [hunk 20]
```diff
@@ -23,54 +23,62 @@
...
-    "@types/webpack": "5.28.1",
+    "@types/d3-hierarchy": "^2.0.0",
+    "@types/d3-selection": "^2.0.0",
+    "@types/d3-shape": "^2.0.0",
+    "@types/jasmine": "^3.6.2",
+    "@types/jasmine-jquery": "^1.5.33",
+    "@types/jquery": "3.5.5",
+    "@types/karma": "^6.3.4",
+    "@types/lodash": "4.14.166",
+    "@types/webpack": "4.41.25",
```
Comment (AleksSavelev) [inline package.json:34]: Why did you revert these changes?

Comment (SanderGi) [reply package.json:34]: I ran the tests and they failed, one of the errors were "function getJSONFixtures() not found on jasmine" so it seemed the configuration was wrong. I tried figuring out what went wrong, but in the end it was easier to just revert the changes

File: package.json (hunk 19 above)
Comment (SanderGi) [reply package.json]: It was already 5.0.2 due to the "^" but I have updated the number explicitly as well: [81189d4](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/81189d443a2bda4f576855676a9396fd4aace44a)

File: package.json (hunk 18 above)
Comment (SanderGi) [reply package.json]: Done: [81189d4](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/81189d443a2bda4f576855676a9396fd4aace44a)

File: karma.conf.ts (hunk 17 above)
Comment (SanderGi) [reply karma.conf.ts]: That was a left over part of the code from when I forked the original repo. I missed it when removing commented out code. It has been removed now: [81189d4](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/81189d443a2bda4f576855676a9396fd4aace44a)

File: package.json (hunk 20 above)
Comment (uve) [reply package.json:34]: @SanderGi getJSONFixtures fail is related to jasmine-query to load capabilities.json file. You can copy the code from [here](https://github.com/microsoft/powerbi-visuals-sankey/blob/main/test/visualTest.ts#L785)

File: test/visualTest.ts [hunk 21]
```diff
@@ -402,14 +605,15 @@ describe("Sunburst", () => {
...
-                });
+                }, 2, DefaultWaitForRender);
         });
     });
 
     describe("Capabilities tests", () => {
         it("all items having displayName should have displayNameKey property", () => {
-            let r = fetch("base/capabilities.json");
-            let jsonData = JSON.stringify(r);
+            jasmine.getJSONFixtures().fixturesPath = "base";
```
Comment (uve) [inline test/visualTest.ts:614]: please revert these changes. Fetch is the default way to load capabilities file

**Review by uve — CHANGES_REQUESTED** [review]

File: stringResources/en-US/resources.resjson [hunk 22]
```diff
@@ -1,32 +1,26 @@
...
-	"Visual_TextSize": "Text Size",
-	"Visual_Tooltip": "Tooltip",
-	"Visual_DisplayUnits": "Display Units",
-	"Visual_Precision": "Decimal Places" ,
-	"Visual_Description_Legend": "Display legend options",
-	"Visual_Description_Position": "Select the location for the legend",
-	"Visual_Description_Title": "Display a title for legend symbols",
-	"Visual_Description_LegendName": "Title text"
-
+    "Visual_Groups": "Groups",
```
Comment (uve) [inline stringResources/en-US/resources.resjson:2]: Is that change mandatory for your PR?
If not, please revert it

File: test/visualTest.ts [hunk 23]
```diff
@@ -223,168 +314,280 @@ describe("Sunburst", () => {
...
                 dataView,
                 () => {
                     expect(visualBuilder.slices.length).toBe(0);
                     done();
-                });
-            }
+                },
+                2,
+                DefaultWaitForRender);
         });
```
Comment (uve) [inline test/visualTest.ts:327]: drop these 2 arguments in that file

**Review by uve — CHANGES_REQUESTED** [review]

File: stringResources/en-US/resources.resjson (hunk 22 above)
Comment (SanderGi) [reply stringResources/en-US/resources.resjson]: I need to add new localizations for the new inputs and while doing so, I removed the unused localizations and extraneous whitespace. I was asked to rename the category displayname to specify "selected category" in [this comment](https://github.com/microsoft/powerbi-visuals-sunburst/pull/94#issuecomment-1617683260). I believe all the changes are necessary. I've added back the tabs instead of 4-space indents to make the git diff show up nicer: [58ac23c](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/58ac23ca331711844c387f6ad5e4e19bdb717976)

File: test/visualTest.ts (hunk 21 above)
Comment (SanderGi) [reply test/visualTest.ts]: Done: [fafcb2a](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/fafcb2aaf1efb4e1ce60b93d91592fcaa75ec29d)

File: test/visualTest.ts (hunk 23 above)
Comment (SanderGi) [reply test/visualTest.ts]: Done: [fafcb2a](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/fafcb2aaf1efb4e1ce60b93d91592fcaa75ec29d)

File: src/Sunburst.ts [hunk 24]
```diff
@@ -26,6 +26,9 @@
 
 "use strict";
 
+import "core-js/stable";
```
Comment (uve) [inline src/Sunburst.ts:29]: remove core-js and regenerator.
PowerBI doesn't support outdated browsers anymore

**Review by uve — CHANGES_REQUESTED** [review]

File: src/Sunburst.ts (hunk 24 above)
Comment (SanderGi) [reply src/Sunburst.ts]: Done: [7b82e3f](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/7b82e3f4570ec5d0b0f7589707aa4f1dbb900d90)
